src/root/views/DrefApplicationForm/import.ts

```diff
@@ -0,0 +1,573 @@
+  isDefined,
+  isNotDefined,
+  listToGroupList,
+  listToMap,
+  randomString,
+} from '@togglecorp/fujs';
+
+import { dateToDateString } from '#utils/common';
+import { NumericValueOption } from '#types';
+
+function getChildren(node: Node | undefined, name: string) {
+  if (!node || !node.children) {
+    return undefined;
+  }
+
+  const accumulator: Node[] = [];
+
+  if (node.name === name) {
+    accumulator.push(node);
+    return accumulator;
+  }
+
+  node.children.forEach(
+    (child) => {
+      const potentialNode = getChildren(child, name);
+
+      if (potentialNode && potentialNode.length > 0) {
+        accumulator.push(...potentialNode);
+      }
+    }
+  );
+
+  return accumulator;
+}
+
+export async function getImportData(file: File) {
+  const docx = await docxReader.load(file);
+  const body: Node | undefined = docx?.officeDocument?.content?._root?.children?.[1]?.children?.[0];
+  const sdtList = body?.children?.map((ch: Node | undefined) => {
+    const potentialNodeList = getChildren(ch, 'w:sdt');
+
+    return potentialNodeList ?? [];
+  }).flat();
+
+  const importData = sdtList?.map((sdt) => {
+    const alias = getChildren(sdt, 'w:alias');
+    const textNode = getChildren(sdt, 'w:t');
+    const key = alias?.[0]?.attribs?.['w:val'];
+    const value = textNode?.map(
+      t => t.children?.map(
+        tc => tc.data
+      )
+    )?.flat(2)?.join('');
+
+    if (key) {
+      return { key, value };
+    }
+
+    return undefined;
+  });
+
+  return importData?.filter(isDefined);
+}
+
+interface KeyValue {
+  key: string,
+  value: string | undefined,
+}
+
+function getItemsWithMatchingKeys(items: KeyValue[], key: string) {
+  return items.filter(item => item.key.startsWith(key));
+}
+
+function getNumberSafe(str: string | undefined) {
+  if (isNotDefined(str)) {
+    return undefined;
+  }
+
+  if (str === 'Enter number.') {
+    return undefined;
+  }
+
+  const potentialNumber = +str;
+
+  if (Number.isNaN(potentialNumber)) {
+    return undefined;
+  }
+
+  return potentialNumber;
+}
+
+function getBooleanSafe(str: string | undefined) {
+  if (str === 'Yes') {
+    return true;
+  }
+
+  if (str === 'No') {
+    return false;
+  }
+
+  return undefined;
+}
+
+function getDateSafe(str: string | undefined) {
+  if (isNotDefined(str)) {
+    return undefined;
+  }
+
+  const date = new Date(str);
+  if (Number.isNaN(date.getTime())) {
+    return undefined;
+  }
+
+  return dateToDateString(date);
+}
+
+function getStringSafe(str: string | undefined) {
+  if (isNotDefined(str)) {
+    return undefined;
+  }
+
+  const trimmedStr = str.trim();
+
+  if (trimmedStr === '') {
+    return undefined;
+  }
+
+  if (trimmedStr === 'Click or tap here to enter text.') {
+    return undefined;
+  }
+
+  if (trimmedStr === 'Enter indicator title.') {
+    return undefined;
+  }
+
+  if (trimmedStr === 'Name.' || trimmedStr === 'Email.' || trimmedStr === 'Title.' || trimmedStr === 'Phone number.') {
+    return undefined;
+  }
```

Comment:
Let's use a map instead. Also, it would be best if we could add some prefix/suffix on the document itself.